enterprise/server/vmcasfs/vmcasfs.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/casfs"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/container"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/dirtools"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/vsock"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"google.golang.org/grpc"
+
+	vmfspb "github.com/buildbuddy-io/buildbuddy/proto/vmcasfs"
+	libVsock "github.com/mdlayher/vsock"
+	bspb "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var (
+	port = flag.Uint("port", vsock.VMCASFSPort, "The vsock port number to listen on")
+)
+
+type casfsServer struct {
+	cfs      *casfs.CASFS
+	bsClient bspb.ByteStreamClient
+
+	mu sync.Mutex
+	// Context & cancel func for CAS RPS.
+	remoteRPCCtx       context.Context
+	cancelRemoteRPCCtx context.CancelFunc
+}
+
+func NewServer() (*casfsServer, error) {
+	if err := os.Mkdir("/casfs", 0755); err != nil {
+		return nil, err
+	}
+	cfs := casfs.New("/workspace", "/casfs/", &casfs.Options{})
```

Comment:
do these paths have to match what is in goinit/main.go and firecracker/firecracker.go? Any ideas on how we could consolidate them? (maybe pass flags to init and to this func and define the consts in firecracker.go?)